review first: add wave to get realtime weather data (#9)
AdaGold/weather-report: ada-project-docs/wave-04.md

## Wave 4: Calling APIs

> In this wave, we will add a feature that gets the weather of the current displayed city name. In order to get the weather of the city, first we will need to get the latitude and longitude of the city using the [`LocationIQ`](https://locationiq.com/docs) API. We can then use the latitude and longitude with the [`OpenWeather`](https://openweathermap.org/api/one-call-api) API to get current weather data.

Reviewer: ```suggestion
For this wave, we will add a feature that gets the weather of the currently displayed city name. In order to get the weather of the city, first we will need to get the latitude and longitude of the city using the [`LocationIQ`](https://locationiq.com/docs) API. We can then use the latitude and longitude with the [`OpenWeather`](https://openweathermap.org/api/one-call-api) API to get current weather data.
```

> We will need to create accounts and log into `LocationIQ` and `OpenWeather`. We will need to navigate the site to find our API keys and read the docs to learn how to make the correct `GET` request and parse the response.

Reviewer: ```suggestion
First, we will need to create accounts and log into `LocationIQ` and `OpenWeather`. Next, we will navigate the sites to find our API keys and read the docs to learn how to make the correct `GET` requests and parse the responses.
```

### Proxy Servers

> Both `LocationIQ` and `OpenWeather` require API keys. When we are building client side web applications, we don't want to include these API keys in our code as they will be available for anyone to see. To manage our API keys we will use a proxy server.

Reviewer: Since the paragraph above talks about navigating the sites to get the API keys, I think we could remove the first sentence and there wouldn't be confusion.
```suggestion
When we are building client side web applications, we don't want to include our API keys in our code as they will be available for anyone to see. To manage our API keys we will use a proxy server.
```

"A proxy server is a server (or computer system or an application) that acts as an intermediary for requests from clients seeking resources from other servers" [resource](https://medium.com/system-design-blog/what-is-proxy-server-a05b99bf686a#:~:text=A%20proxy%20server%20is%20a,seeking%20resources%20from%20other%20servers). 

### Weather Report Proxy Server

For Weather Report, we provide a [weather report proxy server](https://github.com/adaGold/weather-report-proxy-server) built with Flask. As such, the *Weather Report web app* we are building should make `axios` calls to our *Weather Report proxy server*, which manages our API keys and forwards our requests along to the appropriate 3rd party API. 

> For details on how to run the Weather Report proxy server, see the [Weather Report Proxy Server README](https://github.com/adaGold/weather-report-proxy-server). No changes need to made to the source code of the proxy server. The only work that needs to be done is adding a `.env` file with your API keys.

Reviewer: ```suggestion
For details on how to run the Weather Report proxy server, see the [Weather Report Proxy Server README](https://github.com/adaGold/weather-report-proxy-server). No changes need to be made to the source code of the proxy server. The only work that needs to be done is adding a `.env` file with your API keys.
```

## Wave 4: Calling APIs

> For this wave, we will add a feature that gets the weather of the currently displayed city name. In order to get the weather of the city, first we will need to get the latitude and longitude of the city using the [`LocationIQ`](https://locationiq.com/docs) API. We can then use the latitude and longitude with the [`OpenWeather`](https://openweathermap.org/api/one-call-api) API to get current weather data.

Reviewer 2: I would add some of the information under the "Weather Report Proxy Server" section about the main TODO being making `axios` calls to call `LocationIQ` and `OpenWeather` up here to the intro. 

I think enough information is given overall, but so much new information about proxy servers and CORs is introduced that I could see students getting confused about what we're actually asking them to do.

> # Wave 4: Calling APIs

Reviewer 3: Both times when I started to write my axios calls, VSCode "helpfully" inserted a require call that caused an error
```js
const { default: axios } = require("axios");
```
It could be worth calling out that this might happen and should be removed.

## LocationIQ and OpenWeather

In order to get the weather of the city, we will need to get the latitude and longitude of the city using the [`LocationIQ`](https://locationiq.com/docs) API. We can then use the latitude and longitude with the [`OpenWeather`](https://openweathermap.org/api/one-call-api) API to get current weather data. 

To get started and generate API tokens, create accounts and log into `LocationIQ` and `OpenWeather`.

## Weather Report Proxy Server

For Weather Report, we provide a [weather report proxy server](https://github.com/adaGold/weather-report-proxy-server) built with Flask. 

As such, the *Weather Report web app* we are building should make `axios` calls to our *Weather Report proxy server*, which manages our `LocationIQ` and `OpenWeather` API keys and forwards our requests along to the appropriate 3rd party API. 

For details on how to run the Weather Report proxy server, see the [Weather Report Proxy Server README](https://github.com/adaGold/weather-report-proxy-server). No changes need to be made to the source code of the proxy server. The only work that needs to be done is adding a `.env` file with your API keys.

> While the `GET` requests to `LocationIQ` and `OpenWeather` are already implemented in the proxy server, we should still read the [`LocationIQ`](https://locationiq.com/docs) and [`OpenWeather`](https://openweathermap.org/api/one-call-api) docs to learn how to build the request and parse the response.

Reviewer 3: It's a little unfortunate I can't pass the `units` argument to the proxy server. True, we can convert the resulting kelvin to our preferred temperature range, but I'd rather just tell the server the units I would like to use.
